gmso/core/topology.py

```diff
@@ -1367,6 +1405,34 @@ def iter_sites_by_molecule(self, molecule_tag):
         else:
             return self.iter_sites("molecule", molecule_tag)
 
+    def iter_connections_by_site(self, site, connections=None):
```

Comment:
I have two questions about this method:

1- Is there a reason this is not a private method? 

2- Are there scenarios in which the value of `connections` is something other than the default options you provided (`["bonds", "angles", "dihedrals", "impropers"]`)? If yes, maybe it's worth adding unit tests for such cases.

Reply:
1. I think there are enough use cases for wanting quick access to a site's specific connections to not make it private.

2. I don't think there would be cases where it could be something other than this list. It's really just a list of attribute names in the `Topology` class. Issues could arise if someone has a typo, or uses a singular version (e.g. `bond` instead of `bonds`).  We could throw an error if this happens.